bioio_ome_zarr/writers/metadata.py

```diff
@@ -0,0 +1,329 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+
+# Public NGFF version constants
+OME_NGFF_VERSION_V04 = "0.4"
+OME_NGFF_VERSION_V05 = "0.5"
+
+
+class Axes:
+    """
+    Holds axis metadata for an N-D image, aligned with NGFF 0.5 axes, and
+    renders cleanly for both NGFF 0.4 (Zarr v2) and NGFF 0.5 (Zarr v3).
+
+    Attributes
+    ----------
+    ndim : int
+        Number of dimensions in the image data.
+    names : List[str]
+        Names of each axis (e.g., ["t", "c", "z", "y", "x"]).
+    types : List[str]
+        NGFF axis types for each axis (e.g., "time", "channel", "space").
```

Comment:
would it make sense for time, channel and space to be an enumeration?  Could there be types besides those 3?